EventFilter/CTPPSRawToDigi/src/RawDataUnpacker.cc

```diff
@@ -374,6 +380,43 @@ int RawDataUnpacker::ProcessVFATDataParallel(const uint16_t *buf, unsigned int O
     presenceFlags |= 0x8;
     fd[0] = buf[dataOffset + 8];
   }
+  
+  // get channel data for diamond compact mode
+  if (hFlag == vmDiamondCompact)
+  {
+    for (unsigned int i = 1; (buf[i+1] & 0xFFF0)!= 0xF000; i++) {
```

Comment:
should this not protect against an infinite loop? i guess by checking against maxwords ?